xarray/core/dataarray.py

```diff
@@ -178,13 +207,23 @@ def __init__(self, data, coords=None, dims=None, name=None,
             instead of a copy if possible.
         coords : sequence or dict of array_like objects, optional
             Coordinates (tick labels) to use for indexing along each dimension.
-            If dict-like, should be a mapping from dimension names to the
-            corresponding coordinates. If sequence-like, should be a sequence
-            of tuples where the first element is the dimension name and the
-            second element is the corresponding coordinate array_like object.
-        dims : str or sequence of str, optional
-            Name(s) of the data dimension(s). Must be either a string (only
-            for 1D data) or a sequence of strings with length equal to the
+            The following notations are accepted:
+
+            - mapping {dimension name: array-like}
+            - sequence of tuples (dimension name, array-like)
```

Comment:
The actual format for these tuples are any valid arguments for `xarray.Variable()`. So any of these would work: `(dims, data)`, `(dims, data, attrs)` and `(dims, data, attrs, encoding)`.

For now, it might be better to only enforce `tuple` as the type.